analyses/malawi/notebooks/mwi_ecmwf_monthly_skill.md

```diff
@@ -0,0 +1,946 @@
+ecmwf_country_data_processed_dir = os.path.join(country_data_processed_dir,"ecmwf")
+monthly_precip_exploration_dir=os.path.join(country_data_exploration_dir,"dryspells","monthly_precipitation")
+
+plots_dir=os.path.join(country_data_processed_dir,"plots","dry_spells")
+plots_seasonal_dir=os.path.join(plots_dir,"seasonal")
+
+adm2_bound_path=os.path.join(country_data_raw_dir,config.SHAPEFILE_DIR,parameters["path_admin2_shp"])
+all_dry_spells_list_path=os.path.join(country_data_processed_dir,"dry_spells","full_list_dry_spells.csv")
+monthly_precip_path=os.path.join(country_data_processed_dir,"chirps","seasonal","chirps_monthly_total_precipitation_admin1.csv")
+```
+
+# Determine general skill per leadtime
+
+
+### Read in forecast and observational data
+
+```python
+# #only needed if stats files haven't been computed before and are not on the gdrive
+# #takes several hours
+# processing.compute_stats_per_admin(country)
+```
+
+```python
+da = processing.get_ecmwf_forecast()
+```
+
+```python
+da_lt=processing.get_ecmwf_forecast_by_leadtime()
+```
+
+```python
+da_obs=rioxarray.open_rasterio(chirps_monthly_mwi_path,masked=True)
+#only select the years for which we also identified dry spells
+da_obs=da_obs.sel(time=da_obs.time.dt.year.isin(range(2000,2021)))
+```
+
+```python
+#interpolate forecast data such that it has the same resolution as the observed values
+#using "nearest" as interpolation method and not "linear" because the forecasts are designed to have sharp edged and not be smoothed
+da_lt_interp=da_lt.interp(latitude=da_obs["y"],longitude=da_obs["x"],method="nearest")
+```
+
+Let's take a sample of some of the data to check that it all looks like we would expect. 
+
+```python
+# Slice time and get mean of ensemble members for simple plotting
+start = '2020-06-01'
+# end = '2020-10-31'
+
+rf_list_slice = da_lt.sel(time=start,latitude=da_lt.latitude.values[5],longitude=da_lt.longitude.values[5])
+
+rf_list_slice.dropna("leadtime").plot.line(label='Historical', c='grey',hue="number",add_legend=False)
+rf_list_slice.dropna("leadtime").mean(dim="number").plot.line(label='Historical', c='red',hue="number",add_legend=False)
+plt.show()
+```
+
+#### Compute the measure(s) of forecast skill
+
+We'll compute forecast skill using the ```xskillscore``` library and focus on the CRPS (continuous ranked probability score) value, which is similar to the mean absolute error but for probabilistic forecasts.
+
+```python
+df_crps=pd.DataFrame(columns=['leadtime', 'crps'])
+
+#rainy includes the months to select
+#thresh the thresholds
+subset_dict={"rainy":[12,1,2],"thresh":[170]}
+
+for leadtime in da_lt_interp.leadtime:
+    forecast = da_lt_interp.sel(
+    leadtime=leadtime.values).dropna(dim='time')
+    observations = da_obs.reindex({'time': forecast.time}).precip
+    # For all dates
+    crps = xs.crps_ensemble(observations, forecast,member_dim='number')
```

Comment:
Since we're calculating the CRPS for many different countries now, I've added a method for calculating it [here](https://github.com/OCHA-DAP/pa-anticipatory-action/pull/82/files?file-filters%5B%5D=.md&file-filters%5B%5D=.py&file-filters%5B%5D=.yml#). This script should probably eventually be updated to use it.

Reply:
This is really nice! Yes definitely. I am gonna leave it like this for now if that is alright and then get back to it later.